cpp/src/io/utilities/config_utils.hpp

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cstdlib>
+#include <string>
+
+namespace cudf::io::detail {
+
+/**
+ * @brief Returns the value of the environment variable, or a default value if the variable is not
+ * present.
+ */
+inline std::string getenv_or(std::string const& env_var_name, std::string_view default_val)
+{
+  auto const env_val = std::getenv(env_var_name.c_str());
+  return std::string{(env_val == nullptr) ? default_val : env_val};
+}
+
+namespace nvcomp_integration {
+
+namespace {
+/**
+ * @brief Defines which nvCOMP usage to enable.
+ */
+enum class usage_policy : uint8_t { NONE, STABLE, EXPERIMENTAL };
+
+/**
+ * @brief Get the current usage policy.
+ */
+inline usage_policy get_env_policy()
+{
+  static auto const env_val = getenv_or("LIBCUDF_NVCOMP_POLICY", "STABLE");
+  if (env_val == "NONE") return usage_policy::NONE;
+  if (env_val == "EXPERIMENTAL") return usage_policy::EXPERIMENTAL;
+  return usage_policy::STABLE;
+}
+}  // namespace
+
+/**
+ * @brief Returns true if stable nvCOMP use is enabled.
+ */
+inline bool is_stable_integration_enabled() { return get_env_policy() == usage_policy::STABLE; }
+
+/**
+ * @brief Returns true if experimental nvCOMP use is enabled.
+ *
+ * Note: When experimental use is enabled, stable use is enabled as well
+ */
+inline bool is_experimental_integration_enabled()
+{
+  return is_stable_integration_enabled() or get_env_policy() == usage_policy::EXPERIMENTAL;
```

Comment:
`is_stable_integration_enabled` is currently not used, so I can only infer its semantics. But I'd assume, that `stable` should be a subset of `experimental` not the other way round? E.g., let's assume `brotli` would be experimental for nvcomp, then:
```
if (nvcomp_integration::is_experimental_integration_enabled()) {
    brotli_decompress(...);
}else{
    // libcudf code path
}
```
As I understand the conditional above, if the user has chosen `STABLE`, we'd take the code path for experimental nvcomp debrotli, even though the user actually only wanted to use the stable ones?